crypto/evp/evp_lib.c

```diff
@@ -1236,17 +1236,10 @@ EVP_PKEY *EVP_PKEY_Q_keygen(OSSL_LIB_CTX *libctx, const char *propq,
         name = va_arg(args, char *);
         params[0] = OSSL_PARAM_construct_utf8_string(OSSL_PKEY_PARAM_GROUP_NAME,
                                                      name, 0);
-    } else if (OPENSSL_strcasecmp(type, "ED25519") != 0
-               && OPENSSL_strcasecmp(type, "X25519") != 0
```

Comment:
While it's great that the function can now handle externally supplied/unrecognized key type names without requiring specific/additional parameters, I just want to double check it would not cause any issues down the line. Could you please double check and ensure it by replying to this comment?

Reply:
Can you please give some examples of the issues that you anticipate here?

Reply:
The only real issue I see is if a new algorithm is added to some provider that needs a parameter for its keygen to work. Until this function is updated to provide that parameter the keygen call is going to fail for that algorithm. But that's no worse than the current situation (it fails now). At least with this change, algorithms that don't require a parameter work immediately.

Reply:
It's a pity that we decided to uas stdarg here...  *or* that we chose not to pass along the `va_list` (this is entirely possible, as its ABI shouldn't change between builds, and it's a very mature API at this point).

Reply:
Time for `EVP_PKEY_Q_keygen_ex()`?